src/components/Pagination/Pagination.tsx

```diff
@@ -1,42 +1,209 @@
-    nextPage?: React.ReactNode;
-    /** Left hand element that is the previous handler, typically a button */
-    previousPage?: React.ReactNode;
+    /** The total number of pages. */
+    pageCount: number;
+    /** The current page selected. */
+    currentPage: number;
+    /** The method to callback when an item is selected. Passes the selected page to the consuming app as an argument. */
+    onPageChange?: (selected) => void;
 }
 
-export default function Pagination(
-    props: React.PropsWithChildren<PaginationProps>
-) {
-    const {
-        blockName,
-        children,
-        className,
-        modifiers,
-        nextPage,
-        previousPage,
-    } = props;
-
-    const pagination__base_class = "pagination";
-
-    return (
-        <nav
-            aria-label="Pagination"
-            className={bem(pagination__base_class, modifiers, blockName, [
-                className,
-            ])}
-        >
-            {previousPage}
-            {children}
-            {nextPage}
-        </nav>
-    );
+/** Pagination component that provides a list of page items */
+export default class Pagination extends React.Component<PaginationProps> {
+    constructor(props: PaginationProps) {
+        super(props);
+    }
+
+    previousPage = (evt) => {
+        const { currentPage } = this.props;
+        evt.preventDefault ? evt.preventDefault() : (evt.returnValue = false);
+        if (currentPage > 1) {
+            this.selectPage(evt, currentPage - 1);
+        }
+    };
+
+    nextPage = (evt) => {
+        const { currentPage } = this.props;
+        const { pageCount } = this.props;
+
+        evt.preventDefault ? evt.preventDefault() : (evt.returnValue = false);
+        if (currentPage < pageCount) {
+            this.selectPage(evt, currentPage + 1);
+        }
+    };
+
+    selectPage = (evt, item) => {
+        const { currentPage } = this.props;
+
+        evt.preventDefault ? evt.preventDefault() : (evt.returnValue = false);
+        if (currentPage === item) return;
+
+        // Run the callback with the new selected item:
+        this.runCallback(item);
+    };
+
+    runCallback = (selectedItem) => {
+        if (
+            typeof this.props.onPageChange !== "undefined" &&
+            typeof this.props.onPageChange === "function"
+        ) {
+            this.props.onPageChange(selectedItem);
+        }
+    };
+
+    getPageElement(item) {
+        const { currentPage } = this.props;
+
+        let pageAttributes = {
+            "aria-label": null,
+            onClick: (evt) => this.selectPage(evt, item),
+            role: "button",
+            tabIndex: 0,
+        };
+
+        pageAttributes["aria-label"] = item ? item : null;
+
+        return (
+            <li key={item}>
+                <Link
+                    attributes={{ ...pageAttributes }}
+                    className={currentPage === item ? "selected" : null}
+                    href="#"
```

Comment:
Does this being specified here prevent users from updating the route in the url bar?

Reply:
It doesn't affect the url bar. There is an event handler on the click event that prevents that.